File: Week4_Arrays/move-zeros.py
class Solution:
    # Simpler version of sort colors
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # Pointer to store number of non-zeros seen during traversal
        # non zeros used to push zeros to the back
        nonZeroPtr = 0
        for i in range(len(nums)):
            num = nums[i]
            # if number not 0, then perform swapping operation
            if num != 0:
                nums[i], nums[nonZeroPtr] = nums[nonZeroPtr], nums[i]
                # shift non-zeros pointer to the right
                nonZeroPtr += 1

File: Session2_Arrays/move-zeros.java
class Solution {
    // simpler version of sort colors
    public void moveZeroes(int[] nums) {
        /* Pointer to store number of non-zeros seen during traversal */
        int nonZeroPtr = 0;
        /* For-loop to push zeros to the rear of array */
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            /* If number not 0, perform swapping */
            if (num != 0) {
                int temp = nums[i];
                nums[i] = nums[nonZeroPtr];
                nums[nonZeroPtr] = temp;
                /* shift non-zero pointer to the right */
                nonZeroPtr++;
            }
        }
    }
}